pwa-devdocs/src/venia-ui/reference/components/Shimmer/index.md

```diff
@@ -0,0 +1,179 @@
+The Shimmer component eliminates most of the CLS on a page, which helps improve Google Lighthouse scores.
+
+<!--
+The reference doc content is generated automatically from the source code.
+To update this section, update the doc blocks in the source code
+-->
+
+{% include auto-generated/venia-ui/lib/components/Shimmer/shimmer.md %}
+
+## Accessibility
+
+To maintain accessibility for screen readers, we can pass `aria-live="polite" aria-busy="true"` to the Shimmer component (or an
+element that wraps the Shimmer(s) in a more complex instance).
+
+It's important to then add `aria-live="polite" aria-busy="false"` to the _normal_ component that replaces the shimmer.
+
+### Example
+
+```jsx
+// ....
+import Shimmer from '../path/to/base/Shimmer';
+// ....
+export default () => {
+  // ....
+  return (
+    <Shimmer />
+  );
+};
+```
+
+## Shimmer for Components
+
+When loading data, previously we would return `null` (or a full-screen loader) instead of the actual component. We can now return a
+_**shimmer**_ version of the component, which will take up the same space without relying on data.
+
+Direct use of the `Shimmer` component within _normal_ components should be avoided when possible. If the shimmer is replacing a component
+that is imported, a `.shimmer.js` file should be created for that imported component in its folder and exported in its `index.js`.
+
+### Example
+
+There are 4 critical files for creating a Shimmer component:
+
+* **Main.js** - the main component that has the loading status and would usually return null for the component while loading
+* **SubComponent/index.js** - Previously would only export the main component. Must now export named variable for shimmer component
+* **SubComponent/subComponent** - Same SubComponent as usual
+* **SubComponent/subComponent.shimmer.js** - `.shimmer.js` extension is used for easily identifying that it's a shimmer and
+  the component it's attached to. It can contain complex arrangement of base Shimmer elements, or include other subcomponents Shimmers.
+
+**Main.js**
+
+```jsx
+import React from 'react';
+import SubComponent, { SubComponentShimmer } from '../path/to/SubComponent';
+// ....
+export default () => {
+    const { data, isLoading } = fetchData();
+
+    if (isLoading) {
+        return (
+          <SubComponentShimmer />
+        );
+    }
+
+    if (!data) {
+        return 'No data';
+    }
+
+    return (
+        <SubComponent someValue={data} />
+    );
+};
+// ....
+```
+---
+
+**../path/to/SubComponent/index.js**
+
+```jsx
+export { default } from './subComponent.js';
+// Export named shimmer component
+export { default as SubComponentShimmer } from './subComponent.shimmer.js';
+```
+---
+
+**../path/to/SubComponent/subComponent.js**
+
+```jsx
+import React from 'react';
+import { shape, string } from 'prop-types';
+import { mergeClasses } from '../../path/to/classify';
```

Comment:
I'm not sure if it's worth updating, but we now use `useStyle` instead of `mergeClasses` although it still exists. I'm not sure if other docs have been updated with this change already

Reply:
Got it. I changed it throughout this file and made a note to create a ticket as part of the PWA Documentation Updates epic: https://jira.corp.magento.com/browse/PWA-2397